File: day01-数组-二分查找/704.二分查找/704. 二分查找.py
class Solution(object):
    def search(self, nums, target) -> int:
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        right = len(nums) - 1
        left = 0
        while left <= right:
            mid = (right + left) / 2
            if nums[mid] > target:
                right = mid - 1
            elif nums[mid] < target:
                left = mid + 1
            else:
                return mid
        return -1

File: day01-数组-二分查找/704.二分查找/704. 二分查找.cpp
#include <iostream>;
#include <vector>;
using namespace std;
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int right = nums.size();
        int left = 0;
        while (left < right)
        {
            int mid = (right + left) / 2;
            if (nums[mid] < target)
            {
                left = mid;
            }
            else if (nums[mid] > target)
            {
                right = mid;
            }
            else
            {
                return mid;
            }
        }
        return -1;
    }
};